stm32loader/bootloader.py

```diff
@@ -322,15 +329,21 @@ def get_flash_size(self, device_family):
     def get_uid(self, device_id):
         """Send the 'Get UID' command and return the device UID."""
         uid_address = self.UID_ADDRESS[device_id]
-        uid = self.read_memory(uid_address, 12)
-        return uid
+        if uid_address == self.UID_NOT_SUPPORTED:
+            return self.UID_NOT_SUPPORTED
+        else:
+            uid = self.read_memory(uid_address, 12)
+            return uid
 
     @staticmethod
     def format_uid(uid):
         """Return a readable string from the given UID."""
-        swapped_data = [[uid[b] for b in part] for part in Stm32Bootloader.UID_SWAP]
-        uid_string = "-".join("".join(format(b, "02X") for b in part) for part in swapped_data)
-        return uid_string
+        if uid == -1:
```

Comment:
This, admittedly, is not very clean but since the method is defined static we can not get `self.ID_NOT_SUPPORTED`. Wasn't sure where to define that constant, please advise if you need this changed.

Reply:
You can make it a `@classmethod` to get access to class-level constants.

I think I prefer using `None` instead of `UID_NOT_SUPPORTED` in the `UID_ADDRESS` dict when we know the part does not have a UID address.

And `get_uid` could return either of two constants, `UID_NOT_PRESENT` (value 0) for this current case, and `UID_NOT_SUPPORTED` (value -1) in case the part *might* have a UID, but stm32loader does not know the address for it yet.